MetaMask/metamask-extension
# Fix typed message user rejection code

## Explanation

Recent changes around Signature Controller caused SignedTypedMessage signatures reject with wrong rejection code. 
It will be eventually fixed in https://github.com/MetaMask/core/pull/1377 and https://github.com/MetaMask/metamask-extension/pull/19184 but until then we can patch it to mitigate this issue. 



## Screenshots/Screencaps

Correct error code in test-dapp after rejecting Signed Type signature
<img width="600" alt="image" src="https://github.com/MetaMask/metamask-extension/assets/7644512/5874a9f9-2eba-474e-9b35-2d9208a1b33b">

## Manual Testing Steps

- Go to test dapp https://metamask.github.io/test-dapp/ 
- Try using "Sign Typed Data V4"
- Reject approval on notification
- See `4001` error code in the dapp console

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
